app/models/category.rb

```diff
@@ -0,0 +1,9 @@
+class Category < ApplicationRecord
+  has_many :subcategories, class_name: "Category", foreign_key: "parent_id", dependent: :destroy
+  belongs_to :parent_category, class_name: "Category", optional: true
+
+  has_many :products
```

Comment:
If category is associated to products, then what will happen after category destroy?